chia/util/config.py

```diff
@@ -43,13 +49,15 @@ def config_path_for_filename(root_path: Path, filename: Union[str, Path]) -> Pat
 
 def save_config(root_path: Path, filename: Union[str, Path], config_data: Any):
     path: Path = config_path_for_filename(root_path, filename)
-    tmp_path: Path = path.with_suffix("." + str(os.getpid()))
-    with open(tmp_path, "w") as f:
-        yaml.safe_dump(config_data, f)
-    try:
-        os.replace(str(tmp_path), path)
-    except PermissionError:
-        shutil.move(str(tmp_path), str(path))
+    with FileLock(path.with_suffix(".lock")):
```

Comment:
I like the idea of using filelock - I believe the default is this will never timeout - are we ok with that?

Reply:
I think that should be fine